src/System.CommandLine/Parser.cs

```diff
@@ -250,6 +247,15 @@ internal IReadOnlyCollection<string> NormalizeRootCommand(IReadOnlyCollection<st
             }
 
             return args;
+
+            bool FirstArgMatchesExeName() =>
+                firstArg != null &&
+                (
+                    firstArg.Equals(commandName, StringComparison.OrdinalIgnoreCase) ||
+                    firstArg.Equals($"{commandName}.exe", StringComparison.OrdinalIgnoreCase)
+                    ||
```

Comment:
Minor thing here. Mixing white space for or operators. Consider always putting them on the end of the line or putting them on their own lines.